src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -172,49 +212,156 @@ private BoundPattern BindPropertyPattern(PropertyPatternSyntax node, BoundExpres
+                        arguments = indexer.Arguments;
+                        argumentNamesOpt = indexer.ArgumentNamesOpt;
+                        argumentRefKindsOpt = indexer.ArgumentRefKindsOpt;
+                        break;
                     }
 
-                    var propertyType = property.GetTypeOrReturnType();
-                    pattern = this.BindPattern(syntax.Pattern, null, propertyType, hasErrors, diagnostics);
-                    result.Add(new BoundSubPropertyPattern(syntax, propertyType, property, resultKind, pattern, hasErrors));
-                }
-                else
+                default:
+                    return BadSubpatternMemberAccess(boundMember, implicitReceiver, memberName, diagnostics, hasErrors);
+            }
+
+            if (!hasErrors && !CheckValueKind(boundMember, BindValueKind.RValue, diagnostics))
+            {
+                hasErrors = true;
+                resultKind = LookupResultKind.NotAValue;
+            }
+
+            return new BoundPropertyPatternMember(
+                memberName,
+                boundMember.ExpressionSymbol,
+                arguments,
+                argumentNamesOpt,
+                argumentRefKindsOpt,
+                expanded,
+                argsToParamsOpt,
+                resultKind,
+                boundMember.Type,
+                hasErrors);
+        }
+
+        private BoundExpression BadSubpatternMemberAccess(
+            BoundExpression boundMember,
+            BoundImplicitReceiver implicitReceiver,
+            IdentifierNameSyntax memberName,
+            DiagnosticBag diagnostics,
+            bool suppressErrors)
+        {
+            if (!suppressErrors)
+            {
+                string member = memberName.Identifier.ValueText;
+                switch (boundMember.ResultKind)
                 {
-                    Error(diagnostics, ErrorCode.ERR_NoSuchMember, propName, type, propName.ValueText);
-                    pattern = new BoundWildcardPattern(node, hasErrors: true);
-                    result.Add(new BoundSubPropertyPattern(syntax, CreateErrorType(), null, resultKind, pattern, true));
+                    case LookupResultKind.Empty:
+                        Error(diagnostics, ErrorCode.ERR_NoSuchMember, memberName, implicitReceiver.Type, member);
+                        break;
+
+                    case LookupResultKind.Inaccessible:
+                        boundMember = CheckValue(boundMember, BindValueKind.RValue, diagnostics);
+                        Debug.Assert(boundMember.HasAnyErrors);
+                        break;
+
+                    default:
+                        Error(diagnostics, ErrorCode.ERR_PropertyLacksGet, memberName, member);
+                        break;
                 }
             }
 
-            return result.ToImmutableAndFree();
+            return ToBadExpression(boundMember, LookupResultKind.NotAValue);
```

Comment:
I don't think so. At this point the problem isn't necessarily a problem with the lookup, but rather that we found the wrong kind of thing.